src/hooks/migration.ts

```diff
@@ -27,20 +27,27 @@ export const useFixCircleState = (circleId: number | undefined) => {
   const fullCircles = useRecoilValue(rApiFullCircle);
   const [, setCircleIdSource] = useRecoilState(rSelectedCircleIdSource);
   const { fetchCircle } = useApiBase();
+  logger.log(`useFixCircleState, circle id: ${circleId}`);
+
+  const [ready, setReady] = useState(false);
 
   useEffect(() => {
     if (!circleId) return;
 
     if (circleId === recoilValue?.circle.id) {
       logger.log(`circle ids match`);
+      setReady(true);
     } else if (fullCircles.has(circleId)) {
       logger.log(`reusing circle data: ${circleId}`);
       setCircleIdSource(circleId);
+      setReady(true);
     } else {
       logger.log(`fetching circle data: ${circleId}`);
-      fetchCircle({ circleId, select: true });
+      fetchCircle({ circleId, select: true }).then(() => setReady(true));
```

Comment:
this prevents a race condition where a new circle's page starts rendering before the data for it is ready, i.e. it uses the previous circle's data

Reply:
Not sure this really would work with all this recoil stuff, but RQ exposes an isFetching prop that maybe could serve as a similar purpose to hold off on page load until the query is complete?